superset/assets/spec/javascripts/components/ColumnTypeLabel_spec.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { shallow } from 'enzyme';
+
+import ColumnTypeLabel from '../../../javascripts/components/ColumnTypeLabel';
+
+describe('ColumnOption', () => {
+  const defaultProps = {
+    type: 'string',
+  };
+
+  let wrapper;
+  let props;
+  const factory = o => <ColumnTypeLabel {...o} />;
+  beforeEach(() => {
+    wrapper = shallow(factory(defaultProps));
+    props = Object.assign({}, defaultProps);
```

Comment:
I think this pattern for test setup is difficult to read in terms of what props are actually applied and it seems strange to allow tests to modify the `props` object directly / reset them every test . I've also seen several code bases move away from using shared `let` variables because it can result in race conditions in more complex tests.

a more readable and common pattern I've seen is:

```javascript
const props = { ...defaultProps };

describe('MyComponent', () => {
  function setup(overrides) {
    const wrapper = shallow(<MyComponent {...props} {...overrides} />);
    return wrapper;
  }

  it('does something', () => {
    const wrapper = setup(/* { optional: override } */);
    expect(...);
  });

  ...
});

```